pkg/kv/kvpb/errors.proto

```diff
@@ -400,14 +400,24 @@ message OpRequiresTxnError {
 // contain the actual value found.
 message ConditionFailedError {
   optional roachpb.Value actual_value = 1;
-  // HadNwereOriginTimestamp is returned when an OriginTimestamp was
+  // HadNewerOriginTimestamp is returned when an OriginTimestamp was
   // sent in the conditional put, the origin timestamp was newer than
-  // the existing value (or no existing value exists), but the
-  // expected value was mismatched. The caller may choose to try again
-  // with the returned ActualValue.
+  // the existing value's "comparison timestamp" (see comment for
+  // OriginTimestampOlderThan) but the expected value was
+  // mismatched. The caller may choose to try again using the returned
+  // ActualValue as the expected value.
```

Comment:
> If we retry this conditionally with the new value, why did we care about the old value in the first place? In other words, why doesn't LDR just want to skip the value comparison part? Is this just to avoid making CPut more confusing, or is there more to it?

There is more to it: When constructing a batch based on a rangefeed-delivered update to a primary key in the source table, this previous value is what was used to construct any _secondary_ index updates that may be necessary (for instance, you may need to issue deletes for now-invalid secondary index updates).  If the previous value is wrong, then the secondary index updates are also potentially wrong. Thus if we win the origin timestamp comparison but have the wrong previous value we want to abandon this entire batch of updates and construct a new one using the actual previous value (delivered via the error here).  On the happy path where we trust the previous value will usually be correct, this allows us to construct batches without first reading the existing row.